sqlitecluster/SQLiteNode.cpp

```diff
@@ -2492,8 +2495,6 @@ void SQLiteNode::postPoll(fd_map& fdm, uint64_t& nextActivity) {
                             // interrupt that chain in a way that will cause the remote end to think you've had an
                             // error, and start over. So, once a connection is established, we should just use that one
                             // for all communication until it breaks.
-                            peer->reset();
-                            _onDisconnect(peer);
```

Comment:
This is significant even though it's a deletion which I know often get somewhat overlooked in reviews.

It's worth reading the comment above this. The comment says that if we're already connected to a peer and we get a new connection to that peer, we should prefer the earlier connection. However, `peer->reset` and `_onDisconnect` do the opposite, they kill the existing peer connection.

We added the `_onDisconnect` here on purpose:
https://github.com/Expensify/Bedrock/pull/1425

But I think this was a mistake. It does make sense to call `_onDisconnect` if we call `peer->reset`, but I think it makes more sense to call neither of them.

Further, the line just below this comment `STHROW("Peer " + peer->name + " seems already connected.");` will discard the newly connected socket as well. So we won't keep the old socket *or* the new socket in this case.

We should either:

```
STHROW()
```

or:

```
peer->reset()
_onDisconnect()
```

But not both. I think doing the throw and discarding the new socket to keep the old one is best.